aries-site/src/examples/templates/table-customization/components/ColumnSettings.js

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { Box, CheckBoxGroup, List, Tab, Tabs, TextInput } from 'grommet';
+import { Search } from 'grommet-icons';
+
+export const ColumnSettings = ({
+  columns,
+  setColumns,
+  visible,
+  setVisible,
+  ...rest
+}) => {
+
+  const [filteredColumns, setFilteredColumns] = useState(columns);
+  const [search, setSearch] = useState('');
+
+  const filterColumns = (array, searchValue = search) => {
+    
+    let filterResults = array;
+
+    if (searchValue) {
+      filterResults = array.filter(o =>
+        Object.keys(o).some(
+          k =>
+            typeof o[k] === 'string' &&
+            o[k].toLowerCase().includes(searchValue.toLowerCase()),
+        ),
+      );
+    }
+    return filterResults;
+  };
+
+  return (
+    <Box pad="small" elevation="medium">
+      <Tabs {...rest}>
+        <Tab title="Select Columns">
+          <Box pad={{ top: 'xsmall' }}>
+            <TextInput
+              type="search"
+              icon={<Search id="search-cols-icon" />}
+              placeholder="Search columns"
+              value={search}
+              onChange={event => {
+                setSearch(event.target.value);
+                const nextColumns = filterColumns(
+                  columns,
+                  event.target.value,
+                );
+                setFilteredColumns(nextColumns);
+              }}
+            />
+            <CheckBoxGroup
+              options={filteredColumns}
+              valueKey="property"
+              labelKey="header"
+              value={visible}
+              onChange={({ value }) => setVisible(value)}
+            />
+          </Box>
+        </Tab>
+        <Tab title="Order Columns">
+          <Box>
+            <List
```

Comment:
I think this List should only allow reordering of the columns selected in "Select Columns". I had deselected "Name" in the first tab, then went to Order and I wouldn't expect "Name" to appear in this list:
<img width="967" alt="Screen Shot 2021-05-27 at 11 14 45 AM" src="https://user-images.githubusercontent.com/12522275/119877036-6666e000-bedd-11eb-8111-c424066ca9f1.png">

Reply:
The figma shows non-visible columns also. We should probably run this through @vavalos5  and @ericsoderberghp

Reply:
Should only be ordering visible columns.